cdc/sink/mq.go

```diff
@@ -106,22 +108,37 @@ func newMqSink(
 			avroEncoder.SetValueSchemaManager(valueSchemaManager)
 			return avroEncoder
 		}
+	case codec.ProtocolCanal:
+		if !config.EnableOldValue {
+			log.Warn("mqSink View all updates as inserts if enable old value is not enabled")
+		}
+		var forceHkPk bool
+		forceHKeyToPKey, ok := opts["force-handle-key-pkey"]
+		if !ok || forceHKeyToPKey == "false" {
+			forceHkPk = false
+		} else if forceHKeyToPKey == "true" {
+			forceHkPk = true
+		}
```

Comment:
Ref [issue](https://github.com/alibaba/canal/issues/2256#issuecomment-535860942), canal adapter can not support sync data changes which do not have primary key, but in cdc we can regard `handle key` as `primary key` when there is not primary key.